### PR TITLE
Added queries and tests

### PR DESCRIPTION
**Pull request by pc2752**
Currently added for digital documents, functions both with identifier, additional optional arguments and without any arguments.

### CODE REVIEW

File: trompace/queries/document.py [hunk 1]
```diff
@@ -59,8 +59,8 @@ def query_document(identifier: str=None, document_name=None, publisher=None, con
     if language and language not in SUPPORTED_LANGUAGES:
         raise UnsupportedLanguageException(language)
 
-    if identifier:
-        return query_create(identifier, QUERY_DIGITALDOCUMENT, document_name, publisher, contributor, creator,
-                                 source, description, language)
+    if all(v is None for v in locals().values()):
```
Comment (alastair) [inline trompace/queries/document.py]: I don't understand what this is supposed to do, can you explain it? `locals()` hides a lot of intent, so it's a good idea to avoid it unless absolutely necessary

File: tests/data/controlaction/EXPECTED_CONTROLACTION.txt [hunk 2]
```diff
@@ -2,7 +2,7 @@ mutation {
   CreateControlAction(
         name: "Verovio MusicXML Converter"
         description: "MusicXML to MEI conversion"
-        actionStatus: accepted
+        actionStatus: active
```
Comment (alastair) [inline tests/data/controlaction/EXPECTED_CONTROLACTION.txt]: try to separate logical changes into separate commits. this makes it easier to review, and find bugs if necessary

File: trompace/connection.py [hunk 3]
```diff
@@ -13,6 +13,23 @@
     """
     q = {"query": querystr}
     server = config.server_id
+    print(server)
```
Comment (alastair) [inline trompace/connection.py]: be careful about completely copying code. In this case I would have got rid of these unnecessary print messages. We should also improve the error reporting in this case - this method should raise an Exception in the case of an error so that the calling client can make a decision on what to do

File: trompace/queries/__init__.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import json
+
+
+class StringConstant:
```
Comment (alastair) [inline trompace/queries/__init__.py]: there's no reason to copy this file - we should have these definitions in only one place in the library

File: trompace/queries/document.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+    Returns:
+        The response from the server as a disctionary object."""
+    query_to_send = query_document(identifier, document_name, publisher, contributor, creator, subject,
+                             description, language)   
+    resp = submit_query_nasync(query_to_send)
+    return resp
+
+
+if __name__ == "__main__":
```
Comment (alastair) [inline trompace/queries/document.py]: not needed

File: trompace/queries/document.py (hunk 1 above)
Comment (pc2752) [reply trompace/queries/document.py]: It's basically to check if any of the input variables is present, by using locals() we can ensure that this continuity remains even if we add more arguments, without having separate checks for each variable.

Comment (pc2752) [reply trompace/queries/document.py]: Basically, this checks if all of the inputs are None. If so, it returns the query required.

File: trompace/queries/person.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    contributor
+    creator
+    source
+    description
+    language
+  }
+}'''
+
+
+def query_artist(identifier: str=None, artist_name=None, publisher=None, contributor=None, creator=None, subject=None,
```
Comment (alastair) [inline trompace/queries/person.py]: let's try and be consistent with terminology here. The object in the CE is a Person, so we should use that word in the client, not artist.

File: trompace/queries/document.py (hunk 5 above)
Comment (pc2752) [reply trompace/queries/document.py]: I thought it was the main function that will be called by the user when sending a query and waiting for a response. Or are you talking about the if __name__ = 'main": bit?

File: trompace/queries/person.py (hunk 6 above)
Comment (pc2752) [reply trompace/queries/person.py]: That makes sense, I was a but confused myself. But I guess this is what was there in the old guidelines for the API. I'll change.

File: tests/data/controlaction/EXPECTED_CONTROLACTION.txt (hunk 2 above)
Comment (pc2752) [reply tests/data/controlaction/EXPECTED_CONTROLACTION.txt]: Can you confirm that active is the same as accepted was before? And also, what should the status be while running a query and after finishing it?

Comment (alastair) [reply tests/data/controlaction/EXPECTED_CONTROLACTION.txt]: We didn't make a clear decision about this. The list of statuses are listed at the bottom of https://schema.org/ActionStatusType. We should decide and update trompa documentation where necessary

File: trompace/queries/document.py [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+import json
+import os
+import asyncio
+
+import websockets
+
+import trompace.config as config
+from trompace.connection import submit_query_nasync, download_file
+from trompace.exceptions import UnsupportedLanguageException, MimeTypeException, QueryException
+from trompace.queries.templates import query_create
```
Comment (alastair) [inline trompace/queries/document.py]: I can't find this templates file

File: trompace/queries/document.py (hunk 1 above)
Comment (alastair) [reply trompace/queries/document.py]: what's the purpose of the if/else here? My feeling is that it's not a good idea to have a different behaviour based on if you have arguments or not, but I can't work out what's happening because I can't find a definition of `query_create`

File: trompace/queries/document.py (hunk 5 above)
Comment (alastair) [reply trompace/queries/document.py]: the main guard - as this is a library there's no need to execute this file as a script

File: trompace/queries/document.py (hunk 7 above)
Comment (pc2752) [reply trompace/queries/document.py]: Added

File: trompace/queries/document.py (hunk 1 above)
Comment (pc2752) [reply trompace/queries/document.py]: If there are no arguments, it would resort to a query which queries all documents. If there is an arguments, it needs to be passed to the query, unfortunately, sending a blank "()" does not work, hence the two queries.

Comment (alastair) [reply trompace/queries/document.py]: ah, right. How about adding the () around the parameters in `query_create` before adding to the template? That way it could be either an empty string, or a string that is formatted `(param: value)`.

File: trompace/queries/templates.py [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+
+
+def query_create(identifier: str, query_string: str, name=None, publisher=None, contributor=None, creator=None,
+                    source=None, subject=None, description=None, language=None, coverage=None, date=None,
+                    disambiguatingDescription=None, relation=None, _type=None, _searchScore=None, additionalType=None,
+                    alternateName=None, image=None, sameAs=None, url=None, additionalName=None,
+                    award=None, birthDate=None, deathDate=None, familyName=None, gender=None, givenName=None,
+                    honorificPrefix=None, honorificSuffix=None, jobTitle=None, knowsLanguage=None):
+    """Returns a mutation for updating an object.
```
Comment (alastair) [inline trompace/queries/templates.py]: you've copied this from a different file without updating the documentation. Please be careful doing this. There's no point in putting time into documentation if it's not correct.

File: trompace/queries/templates.py [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+# To be added EntryPoint, ControlAction, PropertyValueSpecification and Property
+from trompace.exceptions import UnsupportedLanguageException
+from . import make_parameters, MUTATION
+from ..constants import SUPPORTED_LANGUAGES
+
+
+
+
+def query_create(identifier: str, query_string: str, name=None, publisher=None, contributor=None, creator=None,
+                    source=None, subject=None, description=None, language=None, coverage=None, date=None,
```
Comment (alastair) [inline trompace/queries/templates.py]: I see a few parameters here which aren't used in the function body. I'm not sure this is a sustainable way of creating these queries, as it means that each time we add a query for a new type we'll have to see what's not here and add it. This could easily make us miss or duplicate items.
How about moving this code up a level (e.g. to the `query_document` method). We could create a dict:
`query_params = {"name": document_name, "publisher": publisher}`, and then have a version of the `make_parameters` method which skips a field if its value is None.

File: trompace/queries/__init__.py (hunk 4 above)
Comment (pc2752) [reply trompace/queries/__init__.py]: Done for this branch, but will have to do this across branches, otherwise we'll get a lot of conflicts.

File: trompace/queries/templates.py (hunk 9 above)
Comment (pc2752) [reply trompace/queries/templates.py]: Done

File: trompace/connection.py (hunk 3 above)
Comment (pc2752) [reply trompace/connection.py]: Done, but I believe this will be shifted to the trompa processing library.

File: trompace/queries/document.py (hunk 1 above)
Comment (pc2752) [reply trompace/queries/document.py]: Done, both here and for person.

**Comment by pc2752** [comment]
Made changes and per discussion.
